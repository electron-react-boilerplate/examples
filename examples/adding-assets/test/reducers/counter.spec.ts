import { AnyAction } from 'redux';
import counterReducer, {
  increment,
  decrement,
} from '../../app/features/counter/counterSlice';

describe('reducers', () => {
  describe('counter', () => {
    it('should handle initial state', () => {
      expect(counterReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle INCREMENT_COUNTER', () => {
      expect(
        counterReducer({ value: 1 }, { type: increment })
      ).toMatchSnapshot();
    });

    it('should handle DECREMENT_COUNTER', () => {
      expect(
        counterReducer({ value: 1 }, { type: decrement })
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        counterReducer({ value: 1 }, { type: 'unknown' })
      ).toMatchSnapshot();
    });
  });
});
