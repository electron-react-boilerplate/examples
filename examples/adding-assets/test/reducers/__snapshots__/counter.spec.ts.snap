// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle DECREMENT_COUNTER 1`] = `
Object {
  "value": 0,
}
`;

exports[`reducers counter should handle INCREMENT_COUNTER 1`] = `
Object {
  "value": 2,
}
`;

exports[`reducers counter should handle initial state 1`] = `
Object {
  "value": 0,
}
`;

exports[`reducers counter should handle unknown action type 1`] = `
Object {
  "value": 1,
}
`;
